src/printer.ts

```diff
@@ -1567,6 +1567,10 @@ export class PugPrinter {
       val = ` ${val}`;
     }
 
+    if (val === ' ' && this.nextToken?.type === 'indent') {
+      val = val.trimEnd();
+    }
```

Comment:
Awaiting to find a time slot with @ST-DDT to discuss if this is the optimal solution

Reply:
Thx to @ST-DDT we found that the prev applied fix was even incorrect :pray: 
Now it does what the actual expected result should be + it's optimized